doc/basics.md

```diff
@@ -1166,6 +1167,40 @@ std::cout << doc.find_field("k0") << std::endl; // Prints 27
 When the JSON path is the empty string (`""`) applied to a scalar document (lone string, number, Boolean or null), a SCALAR_DOCUMENT_AS_VALUE error is returned because scalar document cannot
 be represented as `value` instances. You can check that a document is a scalar with the method `scalar()`.
 
+JSON Path
+------------
+
+The simdjson library now supports a subset of [JSON Path](https://goessner.net/articles/JsonPath/) through the `at_path()` method, allowing you to reach further into the document in a single call. The subset of JSON path that is implemented is the subset that is trivially convertible into the JSON Pointer format, using `.` to access a field and `[]` to access a specific index.
+
+This implementation relies on `at_path()` converting its argument to JSON Pointer and then calling `at_pointer`, which makes use of [`rewind`](#rewind) to reset the parser at the beginning of the document. Hence, it invalidates all previously parser values, objects and arrays: make sure to consume the values between each call to `at_path`.
+
+Consider the following example:
+
+```c++
+auto cars_json = R"( [
+  { "make": "Toyota", "model": "Camry",  "year": 2018, "tire_pressure": [ 40.1, 39.9, 37.7, 40.4 ] },
+  { "make": "Kia",    "model": "Soul",   "year": 2012, "tire_pressure": [ 30.1, 31.0, 28.6, 28.7 ] },
+  { "make": "Toyota", "model": "Tercel", "year": 1999, "tire_pressure": [ 29.8, 30.0, 30.2, 30.5 ] }
+] )"_padded;
+ondemand::parser parser;
+auto cars = parser.iterate(cars_json);
+cout << cars.at_path("[0].tire_pressure[1]") << endl; // Prints 39.9
+```
+
+A call to `at_path(json_path)` can result in any of the errors that are returned by the `at_pointer` method and if the conversion of `json_path` to json pointer fails, it will lead to an `simdjson::INVALID_JSON_POINTER`error.
+
+```c++
+auto cars_json = R"( [
+  { "make": "Toyota", "model": "Camry",  "year": 2018, "tire_pressure": [ 40.1, 39.9, 37.7, 40.4 ] },
+  { "make": "Kia",    "model": "Soul",   "year": 2012, "tire_pressure": [ 30.1, 31.0, 28.6, 28.7 ] },
+  { "make": "Toyota", "model": "Tercel", "year": 1999, "tire_pressure": [ 29.8, 30.0, 30.2, 30.5 ] }
+] )"_padded;
+ondemand::parser parser;
+auto cars = parser.iterate(cars_json);
+ASSERT_ERROR(cars.at_path("[0].tire_presure[1").get(x), INVALID_JSON_POINTER); // Fails on conversion to json pointer
```

Comment:
typo: tire_presure[1" -> ...[1]"

Why is this a failure? It is because of the typo? It might be helpful to show an example with valid JSON Path that cannot be converted to JSON Pointer. Curious, do we get a detailed error message that explains why conversion failed?

Reply:
For now, there is no detailed error handling when conversion fails and an INVALID_JSON_POINTER is returned.

@lemire do you have any suggestions on how I can provide a detailed error message here? I saw the guideline of not using std::cerr in the contributing.md and I didn't want to introduce the 33rd value in the error type enum.

Reply:
@FranciscoThiesen Got it. I expect there will be a lot of times when a failure is due to JSON Path being valid, but cannot be converted to JSON Pointer. It would be nice to include this example.

Reply:
For now, the conversion to JSON pointer is somewhat naive. The only "check" is that every opening square-bracket should be following by a closing square bracket.

So if the user provides a valid but complicated json path (using slicers, wildcards or other unsupported features), the conversion is likely not going to fail, what will happen is that it will translate to an incorrect / invalid json pointer and fail with the INVALID_JSON_POINTER error.

Reply:
Got it. Thank you for clarifying.

Reply:
@FranciscoThiesen That's right. We disallow printing or aborting within the simdjson library itself. For all sorts of reasons, printing (even to standard error) from within the library is a bad practice. We want the user of the library to be as much in full control as possible, and if we print, that's an event that goes through, possibly to the end user, without control.

What would be allowed is something like an extra function, say `at_path_error_message` which returns a string describing in details the nature of the error.

That is, we typically want to optimize for the case where the request is succesful and merely return an error code. This is cheap. In case of an error and when the user wants more details (which should be an exceptional event), we can have a different code path that does much more work... The key point is that it should be outside any hot path.

So, for example, when parsing invalid JSON, we just return an error code... but we then provide tools so that the user can pinpoint where the error occured.

In this instance here, a more detailed error reporting could be worked out later, possibly.

Reply:
Ok, I can work the at_path_error_message details in a follow-up PR!

Thanks for clarifying the rationale of disallowing printing/aborting!